pkg/events.go

```diff
@@ -365,37 +370,79 @@ func (octopus *EventOctopus) EventPublisher(clientID string) (IEventPublisher, e
 func (octopus *EventOctopus) eventStoreClient() error {
 	logrus.Tracef("Connecting to Stan-Streaming server @ nats://localhost:%d", octopus.Config.NatsPort)
 
-	sc, err := octopus.Client("event-store")
+	sc, err := octopus.Client(ClientID)
 	if err != nil {
 		return err
 	}
 	// Subscribe with manual ack mode
-	// todo store Subscription?
 	_, err = sc.Subscribe(ChannelConsentRequest, func(msg *natsClient.Msg) {
-		event := Event{}
-		// Unmarshal JSON that represents the Order data
-		err := json.Unmarshal(msg.Data, &event)
-		if err != nil {
-			logrus.Errorf("Error unmarshalling event: %w", err)
-			return
-		}
+		event := octopus.storeEvent(msg.Data)
+
 		// Handle the message
-		logrus.Debugf("Received event [%d]: %+v\n", msg.Sequence, event)
+		logrus.Debugf("received event [%d]: %+v\n", msg.Sequence, event)
 
-		if err := octopus.SaveOrUpdate(event); err != nil {
-			logrus.WithError(err).Error("Failed to store event", err)
+		if err := msg.Ack(); err != nil {
+			logrus.WithError(err).Fatal("failed to ack message")
 		}
-		_ = msg.Ack() // Manual ACK
 	}, natsClient.DurableName("consent-request-durable"),
-		natsClient.MaxInflight(1),
 		natsClient.SetManualAckMode(),
+		natsClient.StartWithLastReceived(),
+	)
+
+	// Subscribe to error channel
+	_, err = sc.Subscribe(ChannelConsentErrored, func(msg *natsClient.Msg) {
+		event := octopus.storeEvent(msg.Data)
```

Comment:
These handles are identical except for the debug message. Perhaps make the storeEvent the handler?